lib/archethic/db/embedded_impl/chain_index.ex

```diff
@@ -401,15 +421,15 @@ defmodule Archethic.DB.EmbeddedImpl.ChainIndex do
       address = <<curve_id::8, hash_id::8, hash::binary>>
 
       if timestamp < until do
-        do_search_last_address_until(fd, until, address)
+        do_search_last_address_until(fd, until, {address, timestamp})
       else
         cond do
           timestamp == until ->
             :file.close(fd)
-            address
+            {address, timestamp}
 
           timestamp < until ->
```

Comment:
Is this control usefull as it is already done before ?

Reply:
L431 ?

Reply:
Yes, we are doing 
```
if timestamp < until do 
...
else
  cond do 
    timestamp < until -> 
      ...
  end
end
```
the second timestamp < until will never be true

Reply:
ok